--- C:\Users\bartl\AppData\Local\nvim\docs\ROADMAP\PullRequests\cmp\old.lua	2025-12-18 01:07:01.000000000 +0100
+++ C:\Users\bartl\AppData\Local\nvim\docs\ROADMAP\PullRequests\cmp\FIXED.lua	2025-12-18 01:03:19.000000000 +0100
@@ -32,12 +32,30 @@
 ---Mapping preset configuration.
 mapping.preset = {}
 
 ---Mapping preset insert-mode configuration.
 mapping.preset.insert = function(override)
   return merge_keymaps(override or {}, {
+    ['<Tab>'] = {
+      i = function(fallback)
+        local cmp = require('cmp')
+        if cmp.visible() then
+          cmp.select_next_item()
+        else
+          vim.api.nvim_feedkeys(vim.api.nvim_replace_termcodes('<C-V><Tab>', true, false, true), 'n', false)
+        end
+      end,
+      s = function(fallback)
+        local cmp = require('cmp')
+        if cmp.visible() then
+          cmp.select_next_item()
+        else
+          vim.api.nvim_feedkeys(vim.api.nvim_replace_termcodes('<C-V><Tab>', true, false, true), 'n', false)
+        end
+      end,
+    },
     ['<Down>'] = {
       i = mapping.select_next_item({ behavior = types.cmp.SelectBehavior.Select }),
     },
     ['<Up>'] = {
       i = mapping.select_prev_item({ behavior = types.cmp.SelectBehavior.Select }),
     },
@@ -86,12 +105,13 @@
     ['<Tab>'] = {
       c = function()
         local cmp = require('cmp')
         if cmp.visible() then
           cmp.select_next_item()
         else
-          cmp.complete()
+          vim.api.nvim_feedkeys(vim.api.nvim_replace_termcodes('<C-V><Tab>', true, false, true), 'n', false)
         end
       end,
     },
     ['<S-Tab>'] = {
       c = function()
