--- C:\Users\Bernhard\AppData\Local\nvim\patches\nvchad\tabufline\lazyload\old.lua	2025-12-18 01:21:59.000000000 +0100
+++ C:\Users\Bernhard\AppData\Local\nvim-data\ui\lua\nvchad\tabufline\lazyload.lua	2025-12-11 16:11:34.000000000 +0100
@@ -30,14 +30,16 @@
         table.insert(bufs, args.buf)
       end
     end
 
     -- remove unnamed buffer which isnt current buf & modified
     if args.event == "BufAdd" then
+      if #bufs > 0 and api.nvim_buf_is_valid(bufs[1]) then
       if #api.nvim_buf_get_name(bufs[1]) == 0 and not get_opt("modified", { buf = bufs[1] }) then
         table.remove(bufs, 1)
+        end
       end
     end
 
     vim.t.bufs = bufs
   end,
 })
